main.cpp

```diff
@@ -422,126 +349,91 @@ void triple(std::unique_ptr<T> &heapNumber)
     *heapNumber *= 3;
 }
 
+template<typename T>
+void cube(std::unique_ptr<T> &heapNumber)
+{
+    // previously *heapNumber *= ( (*heapNumber) * (*heapNumber) ) but this doesnt create a temporary for assignment 
+    *heapNumber = ( (*heapNumber) * (*heapNumber) * (*heapNumber) );
+}
+
```

Comment:
All of those `*` are hard to look at.  This is much friendlier:
`auto& v = *heapNumber;   v = v * v * v;`
- [ ] revise

Reply:
That is much cleaner, thanks!